### PR TITLE
Packer best practices and networking documentation

### PR DESCRIPTION
**Pull request by tpdownes**
This PR adds a general overview for requirements and best practices to build custom images using the Toolkit and Packer.

### Submission Checklist

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cloud HPC Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: docs/image-building.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+deployment_groups:
+- group: primary
+  modules:
+  - id: network1
+    source: modules/network/vpc
+
+- group: packer
+  modules:
+  - id: custom-image
+    source: github.com/SchedMD/slurm-gcp//packer?ref=5.7.4&depth=1
```
Comment (nick-stroud) [inline docs/image-building.md:169]: I am not familiar with the `depth` argument. What is this doing?

**Review by nick-stroud — APPROVED** [review]
Resolve comments. Re-review not needed.

File: docs/image-building.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+- [GitHub-hosted modules and packages](https://github.com/tpdownes/hpc-toolkit/blob/packer_documentation/modules/README.md#github-hosted-modules-and-packages)
+- [GitHub-hosted Packer modules](https://github.com/tpdownes/hpc-toolkit/blob/packer_documentation/modules/README.md#github-hosted-packer-modules)
+
+In particular, the Toolkit recommends using double-slash (`//`) notation to
+identify the root of the git repository. Doing so will ensure that the Packer
+module has access to all files within the repository even if it is located in
+a subdirectory.
+
+> **_NOTE:_** If you fail to use `//` notatation when referring to the [SchedMD
+> packer module][schedmd-packer], it will fail to execute because it refers
```
[the remark this answers is not in the document]
Comment (tpdownes) [reply docs/image-building.md:153]: Fixed.

File: docs/image-building.md (hunk 1 above)
Comment (tpdownes) [reply docs/image-building.md:171]: Mentioned here (graf or 2 down): https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/main/modules/README.md#github-hosted-modules-and-packages

TL;DR: for tags and branches (not commit hashes): you can use this and git will only fetch that many commits before checkout-ing your reference. In theory and hopefully practice: faster, less space
